firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,103 @@
+        THEN("the input status is equal to ok")
+        THEN("the final input status is equal to invalid_length")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled with input data")
+    WHEN("the input data exceeds bounds after output is called, without delimiter")
+        THEN("the input status is equal to ok")
+      THEN("the output status is equal to waiting")
+      THEN("the final input status is equal to invalid_length")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled with input data")
+    WHEN("Another byte is given as input")
+      THEN("the final input status should be invalid_length")
+
+    WHEN("output is called after the input status returns invalid length")
+      THEN("the output status is equal to invalid_length")
+
+    WHEN("the input receives delimeter")
+      THEN("input status is equal to output_ready")
+      THEN("the output status is equal to ok")
+      THEN("the final input status is equal to ok")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled completely with input data")
+    WHEN("delimeter is passed after the buffer is full")
+        THEN("the initial status is equal to ok")
+      THEN("the final input status is equal to output ready")
+
+  GIVEN("A uint8_t chunk splitter with internal buffer of 256 bytes, with delimeter equal to 0x01")
+    WHEN("output is called after input status is ok")
+      THEN("the output status is equal to waiting")
+
+    WHEN("input receives delimeter equal to 1 byte")
+      THEN("the output status is equal to ok")
+      THEN("the input status shall be output ready")
+
+  GIVEN("A uint8_t chunk splitter with internal buffer of 256 bytes, with delimeter equal to 255 bytes")
+    WHEN("output is called after input status is ok")
+      THEN("the output status is equal to waiting")
+    WHEN("input receives delimeter equal to 1 byte")
+      THEN("the input status shall be output ready")
+      THEN("the output status is equal to ok")
+
+  GIVEN("A uint8_t chunk splitter with internal buffer of 256 bytes, with include_delimeter bool equal to true")
+    WHEN("output is called after input status is ok")
+      THEN("the output status is equal to waiting")
+    WHEN("input receives delimeter equal to 1 byte")
+      THEN("the input status shall be invalid length")
+      THEN("the output status is equal to invalid length")
+
+  GIVEN("A chunk splitter of buffer size equal to 256 bytes, that is filled with input data")
+    WHEN("The output is called after input of 3 chunks")
+        THEN("the initial status is equal to ok")
+        THEN("the initial status is equal to ok")
+        THEN("the initial status is equal to ok")
+      THEN("the output status is equal to waiting")
+
+
+/* ChunkMerger
+
+Scenario: Protocols::ChunkMerger behaves correctly
+
+  GIVEN("A chunk merger of internal buffer size equal to 30 bytes") {
+    WHEN("the input data given is within bounds")
```

Comment:
I think we need to be more specific about what kind of data is within bounds or out of bounds. For example:

GIVEN a ChunkMerger and 256-byte input data
WHEN the input data is passed as a partially full buffer to transform
THEN transform returns ok status and appends the delimiter byte to the buffer
WHEN the input data is passed as a completely full buffer to transform
THEN transform returns invalid_length status